### PR TITLE
Zach fastsent

### PR DESCRIPTION
**Pull request by zachmayer**
Rebase of #612, with modified `setup.py` + the `.c` version of the `.pyx` file.

### CODE REVIEW

**Comment by zachmayer** [comment]
Rebased against master

File: gensim/models/fastsent.py [hunk 1]
```diff
@@ -0,0 +1,957 @@
...
+# Copyright (C) 2013 Radim Rehurek <me@radimrehurek.com>
+# Licensed under the GNU LGPL v2.1 - http://www.gnu.org/licenses/lgpl.html
+
+
+"""
+FastSent model for sentence representation.
+
+The FastSent algorithm was originally inspired by the C package https://code.google.com/p/fastsent/.
+
+For a blog tutorial on gensim fastsent, which is closely related, visit http://radimrehurek.com/2014/02/fastsent-tutorial/
```
Comment (piskvorky) [inline gensim/models/fastsent.py]: This gives 404 -- I don't remember any such blog post?

**Comment by piskvorky** [comment]
Thanks, it's looking up :)

The Travis tests are still failing though; I'll let @tmylk handle the next steps and review.

File: gensim/models/fastsent.py (hunk 1 above)
Comment (zachmayer) [inline gensim/models/fastsent.py]: Hmm, this is from the original PR: https://github.com/piskvorky/gensim/pull/612/files

Perhaps they meant to link to this post? http://rare-technologies.com/word2vec-tutorial/

**Comment by tmylk** [comment]
@zachmayer  Thank you for attempting to rebase this. The reason that this code lives as a PR is because it is hard to maintain in this form. Refactoring of both word2vec and fastsent is needed to avoid code duplication and to make this rebasing easy. It is a hard project waiting for someone who really needs fastsent integrated to gensim.

**Comment by zachmayer** [comment]
@tmylk I'm not sure I'm up for re-factoring this (but I can try to rebase it again).  

@fh295 Would you be up for re-factoring fastsent to duplicate less code from word2vec?

**Comment by tmylk** [comment]
Duplicate of #835

**Pull request closed without merging**